Issue Resolved: Sharing functionalities added!

swastikom opened the pull request:
Sharing feature #2

With these changes, each joke will have three sharing buttons for Twitter, Facebook, and LinkedIn. When clicked, they will open the respective social media platforms with pre-filled sharing content, such as the joke text or the current page URL.

danicunhac reviewed (changes requested):
Thank for your contribution @swastikom! I'd like to note a few things:
1. The library you used for the icons is not installed. Could you add it to package.json?
2. It throws an error saying `window` is not defined when trying to get the page url.

Once again, thank you for your contribution, can't wait to hear back from you.

swastikom commented:
Hallo there,

As you said previously in the updated code, 

1. I have used 'react-icons'  library for the icons and also added it to package.json.

2. As per the second issue you mentioned that It throws an error saying window is not defined when trying to get the page url. I think this is happening because, the window object is not available in the server-side rendering context of Next.js. The window object is specific to the browser environment and cannot be accessed during the server-side rendering phase. To resolve this issue, I have modified the code to conditionally render the share links only on the client-side. In the updated code, I've used the 'isBrowser' variable to conditionally render the share links only when the code is running in the browser environment. This way, the 'window.location.href' will only be accessed when it's available, which may prevent the error you were encountering during server-side rendering.

Please let me know if there is any further issue.

danicunhac commented:
> Hallo there,
> 
> As you said previously in the updated code,
> 
> 1. I have used 'react-icons'  library for the icons and also added it to package.json.
> 2. As per the second issue you mentioned that It throws an error saying window is not defined when trying to get the page url. I think this is happening because, the window object is not available in the server-side rendering context of Next.js. The window object is specific to the browser environment and cannot be accessed during the server-side rendering phase. To resolve this issue, I have modified the code to conditionally render the share links only on the client-side. In the updated code, I've used the 'isBrowser' variable to conditionally render the share links only when the code is running in the browser environment. This way, the 'window.location.href' will only be accessed when it's available, which may prevent the error you were encountering during server-side rendering.
> 
> Please let me know if there is any further issue.

I don't think this will work with the server-side rendering we're doing. For this we'll have to get rid of using the `window` property., which I think is okay and we can take another approach here. I went ahead and made some changes to your branch we can move it quicker, feel free to tell me if that doesn't work for you! Thanks for the work, it looks amazing!
